Support events qualified for multiple conditions.

amosbird opened the pull request:
Currently windowFunnel function only takes the first qualified event condition
into account when operating on one event. This patch extends the
ability.

I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

sundy-li commented on dbms/src/AggregateFunctions/AggregateFunctionWindowFunnel.h:
```diff
@@ -215,19 +215,13 @@ class AggregateFunctionWindowFunnel final
-        if (event_level)
-        {
-            this->data(place).add(static_cast<const ColumnVector<UInt32> *>(columns[0])->getData()[row_num], event_level);
+                this->data(place).add(timestamp, i);
```
Thank you, but how to avoid duplicate match? 
Eg.
windowFunnel(4)(timestamp, event <= 2, event <=4, event <=6, event <=8)
There's just one row data with `event = 2` , but it could have the result 4.

sundy-li replied:
I got it, nice solution.

amosbird committed:
Support events qualified for multiple conditions.

Currently windowFunnel function only take the first qualified condition
into account when operating on one event. This patch extends the
ability.

VadimPE commented:
@amosbird Your code has problems with test(00632_aggregation_window_funnel). I try to fix it.

amosbird commented:
@VadimPE Hi, I didn't see failures in test 678 and there isn't a related commit in recent commit log. What problem is it?

VadimPE commented:
In this case
`CREATE TABLE test.remote_test(uid String, its UInt32,  action_code String, day Date) ENGINE = MergeTree(day, (uid, its), 8192);`
`INSERT INTO test.remote_test SELECT toString(number) AS uid, number % 3 AS its, toString(number % 3) AS action_code, '2000-01-01' FROM system.numbers LIMIT 10000;`
`SELECT level, COUNT() FROM (SELECT uid, windowFunnel(3600)(toUInt32(its), action_code != '', action_code = '2') AS level FROM remote('127.0.0.{2,3}', test.remote_test) GROUP BY uid) GROUP BY level;`

old version: 1 10000
your version: 1 - 6667
                      2 - 3333

amosbird commented:
Well that's expected. The correct output should be
```
1 - 6667
2 - 3333
```

Old version doesn't work when the some filter includes others, and this is what my patch meant to fix.

VadimPE commented:
@amosbird #3032
